lib/languageclient.ts

```diff
@@ -91,16 +92,32 @@ export class LanguageClientConnection extends EventEmitter {
   }
 
   /**
-   * Public: Register a callback for a custom message.
+   * Public: Register a callback for a custom notification
    *
    * @param method A string containing the name of the message to listen for.
    * @param callback The function to be called when the message is received.
    *   The payload from the message is passed to the function.
    */
-  public onCustom(method: string, callback: (obj: object) => void): void {
+  public onCustomNotification(method: string, callback: (obj: object) => void): void {
     this._onNotification({ method }, callback)
   }
 
+  // @deprecated Use onCustomNotification method instead
+  public onCustom(method: string, callback: (obj: object) => void): void {
+    this.onCustomNotification(method, callback)
+  }
+
+  /**
+   * Public: Register a callback for a custom request
+   *
+   * @param method A string containing the name of the message to listen for.
+   * @param callback The function to be called when the message is received.
+   *   The payload from the message is passed to the function.
+   */
+  public onCustomRequest(method: string, callback: (obj: object) => Promise<object>): void {
```

Comment:
Is it ok or the type should be `callback: (obj: object) => Promise<object | void>`?

Reply:
If the return type is not needed all the time, it is OK to change it to `object | void`.